src/medCore/database/medDatabaseNonPersistentImporter.cpp

```diff
@@ -309,15 +309,18 @@ QString medDatabaseNonPersistentImporter::ensureUniqueSeriesName ( const QString
             seriesNames << sname;
     }
 
-    QString originalSeriesName = seriesName;
     QString newSeriesName = seriesName;
     int suffix = 0;
 
-    while (seriesNames.contains(newSeriesName))
+    if (seriesName == "")
     {
-        // it exist
-        suffix++;
-        newSeriesName = originalSeriesName + " (copy " + QString::number(suffix) + ")";
+        newSeriesName = "UnnamedSeries";
+    }
+
+    suffix = seriesNames.filter(newSeriesName).count();
+    if (suffix > 0)
+    {
+        newSeriesName = newSeriesName + " (copy " + QString::number(suffix) + ")";
```

Comment:
This doesn't work as expected. Suppose you add three unnamed series to the database. They will be named _UnnamedSeries_, _UnnamedSeries (copy 1)_ and _UnnamedSeries (copy 2)_. Now delete the second one (copy 1). That leaves two unnamed series in the DB, which means ```seriesNames.filter(newSeriesName).count()``` returns 2. If you import a new unnamed series, it will get called _UnnamedSeries (copy 2)_, resulting in two series with the same name.